sale_order_line_description/models/sale_order_line.py

```diff
@@ -16,6 +16,8 @@ def product_id_change(self):
         if (self.user_has_groups(
                 'sale_order_line_description.'
                 'group_use_product_description_per_so_line') and
-                self.product_id.description_sale):
-            self.name = self.product_id.description_sale
+                self.product_id.description_sale) and self.order_id.partner_id:
```

Comment:
You shouldn't put the partner as condition here, but only for filling the lang context retrieving the field contents. If no partner, no context, but field must be retrieved.

Reply:
I added the partner to avoid the error mentioned by @aheficent - will remove it .

Reply:
No, you have to do a middle solution.

Reply:
sorry but I tried to squash my commits but I failed to do.
How to squash them ? sorry for that .